pallets/pool-system/src/impls.rs

```diff
@@ -70,6 +73,245 @@ impl<T: Config> PoolInspect<T::AccountId, T::CurrencyId> for Pallet<T> {
 	}
 }
 
+impl<T: Config> PoolMutate<T::PoolId> for Pallet<T> {
+	type AccountId = T::AccountId;
```

Comment:
One minor thing, sorry for the annoyance. But Substrate usually has the `AccountId` as an generic, as it makes no sense to have different `AccountId`s over the runtime. At least my guess, why it is a generic. 

For us, the same applies for the `PoolId`. 

So could you change the trait to take two generics `PoolMutate<AccountId, PoolId>` rather than just one?